### PR TITLE
util perpendicular_distance: Fix division by zero when segment is just a single point

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/util/__init__.py [hunk 1]
```diff
@@ -147,27 +147,37 @@ def rect_intersect(ra, rb):
...
+    """
+    Computes the perpendicular distance between a segment and a point (in 2D space).
+    start (float, float): beginning of the segment
+    end (float, float): end of the segment
+    point (float, float): point anywhere in space
+    return (0 <= float): distance
+    """
     x1, y1 = start
     x2, y2 = end
     x3, y3 = point
```
Comment (Sabrina85) [inline src/odemis/util/__init__.py]: I would suggest to rename it to x_start etc. as it is easier to read the code in between.

File: src/odemis/util/test/util_test.py [hunk 2]
```diff
@@ -144,6 +145,32 @@ def test_simple(self):
             o = util.almost_equal(*i)
             self.assertEqual(o, eo, "Failed to get correct output for %s" % (i,))
 
+
+class PerpendicularDistanceTestCase(unittest.TestCase):
+
```
Comment (Sabrina85) [inline src/odemis/util/test/util_test.py]: doc string what that test does

File: src/odemis/util/__init__.py [hunk 3]
```diff
@@ -147,27 +147,37 @@ def rect_intersect(ra, rb):
 
 
 def perpendicular_distance(start, end, point):
-    """ Return the perpendicular distance between the line described by start and end, and point """
-
+    """
+    Computes the perpendicular distance between a segment and a point (in 2D space).
```
Comment (Sabrina85) [inline src/odemis/util/__init__.py]: what is a segment?

File: src/odemis/util/test/util_test.py [hunk 4]
```diff
@@ -144,6 +145,32 @@ def test_simple(self):
...
+    def test_simple(self):
+        a = (0, 0)
+        b = (1, 0)
+        # Follow a segment at 1 unit away
+        for e in ((0, 1), (0.5, 1), (1, 1)):
+            dist = perpendicular_distance(a, b, e)
+            self.assertAlmostEqual(dist, 1)
+
+        # Follow a segment on the original segment itselft => dist == 0u
```
Comment (Sabrina85) [inline src/odemis/util/test/util_test.py]: point or segment?

File: src/odemis/util/test/util_test.py [hunk 5]
```diff
@@ -144,6 +145,32 @@ def test_simple(self):
             o = util.almost_equal(*i)
             self.assertEqual(o, eo, "Failed to get correct output for %s" % (i,))
 
+
+class PerpendicularDistanceTestCase(unittest.TestCase):
+
+    def test_simple(self):
+        a = (0, 0)
```
Comment (Sabrina85) [inline src/odemis/util/test/util_test.py]: add a doc string as in next method maybe?

File: src/odemis/util/__init__.py (hunk 1 above)
Comment (pieleric) [reply src/odemis/util/__init__.py]: Actually, I prefer not to change the names, as it'd hide the changes required to fix the actual bug.
Ideally, the function readability could be improved, indeed, but in a separate commit (which I'm too lazy to do ;-) ). Maybe writing just `start[0]` and `start[1]` would be as readable.

File: src/odemis/util/test/util_test.py (hunk 2 above)
Comment (pieleric) [reply src/odemis/util/test/util_test.py]: As the name itself implies that it's a test case for "PerpendicularDistance", and I don't think there is much to add to that, I didn't write a docstring.

File: src/odemis/util/__init__.py (hunk 3 above)
Comment (pieleric) [reply src/odemis/util/__init__.py]: Actually, it's just coming from my broken english. Apparently, the official name is "line segment" (as defined in Wikipedia). => Fixed.

File: src/odemis/util/test/util_test.py (hunk 4 above)
Comment (pieleric) [reply src/odemis/util/test/util_test.py]: It's a point... which moves along a line segment.
=> I've updated the comment to make it clearer.

**Comment by pieleric** [comment]
Now updated.

File: src/odemis/util/__init__.py (hunk 1 above)
Comment (Sabrina85) [reply src/odemis/util/__init__.py]: :D

**Pull request merged**